File: Proc/src/hello_thread.c
#include "stdio.h"
#include "pthread.h"

void * callback(void * ignored)
{
	printf("Hello from a thread!\n");
	return NULL;
}

int main()
{
	printf("About to create a thread\n");
	pthread_t thread;
	if(pthread_create(&thread, NULL, callback, NULL))
	{
		fprintf(stderr, "Error creating thread\n");
		return 1;
	}

	if(pthread_join(thread, NULL))
	{
		fprintf(stderr, "Error joining thread\n");
		return 1;
	}

	printf("Thread finished\n");
	return 0;
} 

File: Proc/src/HelloThread.java
public class HelloThread implements Runnable
{
	public void run()
	{
		System.out.println("Hello from a thread!");
	}

	public static void main(String args[])
	{
		System.out.println("About to create a thread");
		Thread thread = new Thread(new HelloThread());
		thread.start();

		try
		{
			thread.join();
		}
		catch (InterruptedException e)
		{
			System.err.println("Error in joining thread");
			return;
		}
		System.out.println("Thread finished");
	}
} 
